src/test/SSTORE2.t.sol

```diff
@@ -82,49 +93,62 @@ contract SSTORE2Test is DSTestPlus {
         startIndex = bound(startIndex, 0, testBytes.length);
 
-        if (startIndex > endIndex) return;
+        if (startIndex > endIndex) (startIndex, endIndex) = (endIndex, startIndex);
 
         assertBytesEq(
             SSTORE2.read(SSTORE2.write(testBytes), startIndex, endIndex),
             bytes(testBytes[startIndex:endIndex])
         );
     }
 
-    function testFailReadInvalidPointer(address pointer) public view {
-        if (pointer.code.length > 0) revert();
+    function testReadInvalidPointer(address pointer) public {
+        if (pointer.code.length > 0) return;
 
+        vm.expectRevert(stdError.arithmeticError);
         SSTORE2.read(pointer);
     }
 
-    function testFailReadInvalidPointerCustomStartBound(address pointer, uint256 startIndex) public view {
-        if (pointer.code.length > 0) revert();
+    function testReadInvalidPointerCustomStartBound(address pointer, uint256 startIndex) public {
+        if (pointer.code.length > 0) return;
 
+        vm.expectRevert(stdError.arithmeticError);
         SSTORE2.read(pointer, startIndex);
     }
 
-    function testFailReadInvalidPointerCustomBounds(
+    function testReadInvalidPointerCustomBounds(
         address pointer,
         uint256 startIndex,
         uint256 endIndex
-    ) public view {
-        if (pointer.code.length > 0) revert();
+    ) public {
+        startIndex = bound(startIndex, pointer.code.length, type(uint256).max - 1);
+        endIndex = bound(endIndex, pointer.code.length, type(uint256).max - 1);
+        if (startIndex > endIndex) (startIndex, endIndex) = (endIndex, startIndex);
 
+        vm.expectRevert("OUT_OF_BOUNDS");
         SSTORE2.read(pointer, startIndex, endIndex);
     }
 
-    function testFailWriteReadCustomStartBoundOutOfRange(bytes calldata testBytes, uint256 startIndex) public {
-        startIndex = bound(startIndex, testBytes.length + 1, type(uint256).max);
+    function testWriteReadCustomStartBoundOutOfRange(bytes calldata testBytes, uint256 startIndex) public {
+        if (testBytes.length == 0) return;
+        startIndex = bound(startIndex, testBytes.length + 1, type(uint256).max - 1);
 
-        SSTORE2.read(SSTORE2.write(testBytes), startIndex);
+        address pointer = SSTORE2.write(testBytes);
+        vm.expectRevert(stdError.arithmeticError);
+        SSTORE2.read(pointer, startIndex);
     }
 
-    function testFailWriteReadCustomBoundsOutOfRange(
+    function testWriteReadCustomBoundsOutOfRange(
         bytes calldata testBytes,
         uint256 startIndex,
         uint256 endIndex
     ) public {
-        endIndex = bound(endIndex, testBytes.length + 1, type(uint256).max);
+        if (testBytes.length == 0) return;
+        endIndex = bound(endIndex, testBytes.length + 1, type(uint256).max - 1);
+        startIndex = bound(startIndex, testBytes.length + 1, type(uint256).max - 1);
+        if (startIndex > endIndex) (startIndex, endIndex) = (endIndex, startIndex);
```

Comment:
why can't we use the previous code again? because of the plus 1? maybe we just allow overflow

Reply:
Otherwise I believe it errors with an arithmetic

Reply:
this way it ensures it reverts with `"OUT_OF_BOUNDS"`